src/core/connection_pool.c

```diff
@@ -0,0 +1,651 @@
+        for (j = 0; j < RssProcessorCount; j++) {
+            if (RssProcessors[j].ProcIndex == RssConfig->RssIndirectionTable[i]) {
+                break;
+            }
+        }
+        //
+        // This is safe because the RssProcessor array is the same count as the indirection table.
+        //
+        if (j == RssProcessorCount) {
+            CXPLAT_DBG_ASSERT(RssProcessorCount < RssConfig->RssIndirectionTableCount);
+            RssProcessors[RssProcessorCount].ConnectionCount = 0;
+            RssProcessors[RssProcessorCount++].ProcIndex = RssConfig->RssIndirectionTable[i];
+        }
+    }
+
+    CXPLAT_DBG_ASSERT(RssProcessorCount > 0);
+    CXPLAT_DBG_ASSERT(RssProcessorCount <= RssConfig->RssIndirectionTableCount);
+
+    *RssProcInfo = RssProcessors;
+    *RssProcCount = RssProcessorCount;
+    return Status;
+}
+
+static
+_IRQL_requires_max_(DISPATCH_LEVEL)
+QUIC_CONN_POOL_RSS_PROC_INFO*
+QuicConnPoolGetRssProcForTuple(
+    _In_ const CXPLAT_TOEPLITZ_HASH* ToeplitzHash,
+    _In_ const QUIC_ADDR* RemoteAddress,
+    _In_ const QUIC_ADDR* LocalAddress,
+    _In_ QUIC_CONN_POOL_RSS_PROC_INFO* RssProcessors,
+    _In_ uint32_t RssProcessorCount,
+    _In_reads_(RssIndirectionTableCount)
+        uint32_t* RssIndirectionTable,
+    _In_ uint32_t RssIndirectionTableCount
+    )
+{
+    //
+    // Calculate the Toeplitz Hash as if receiving packets from
+    // RemoteAddress to find the RSS processor.
+    //
+    uint32_t RssHash = 0, Offset;
+    CxPlatToeplitzHashComputeRss(
+        ToeplitzHash,
+        RemoteAddress,
+        LocalAddress,
+        &RssHash,
+        &Offset);
+
+    const uint32_t Mask = RssIndirectionTableCount - 1;
+
+    CXPLAT_DBG_ASSERT((RssHash & Mask) < RssIndirectionTableCount);
+
+    uint32_t Index = 0;
+    for (; Index < RssProcessorCount; Index++) {
+        if (RssProcessors[Index].ProcIndex == RssIndirectionTable[RssHash & Mask]) {
+            break;
+        }
+    }
+
+    CXPLAT_DBG_ASSERT(Index < RssProcessorCount);
+    return &RssProcessors[Index];
+}
+
+static
+_IRQL_requires_max_(PASSIVE_LEVEL)
+const char*
+QuicConnPoolAllocServerNameCopy(
+    _In_ const char* ServerName,
+    _In_ size_t ServerNameLength
+    )
+{
+    char* ServerNameCopy = CXPLAT_ALLOC_NONPAGED(ServerNameLength + 1, QUIC_POOL_SERVERNAME);
+    if (ServerNameCopy == NULL) {
+        QuicTraceEvent(
+            AllocFailure,
+            "Allocation of '%s' failed. (%llu bytes)",
+            "Server name",
+            ServerNameLength + 1);
+    } else {
+        CxPlatCopyMemory(ServerNameCopy, ServerName, ServerNameLength);
+        ServerNameCopy[ServerNameLength] = 0;
+    }
+
+    return ServerNameCopy;
+}
+
+static
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+QuicConnPoolGetStartingLocalAddress(
+    _In_ QUIC_ADDR* RemoteAddress,
+    _Out_ QUIC_ADDR* LocalAddress
+    )
+{
+    CXPLAT_SOCKET* Socket = NULL;
+    CXPLAT_UDP_CONFIG UdpConfig = { .RemoteAddress = RemoteAddress, };
```

Comment:
```suggestion
    CXPLAT_UDP_CONFIG UdpConfig = { .RemoteAddress = RemoteAddress, 0 };
```
Let's be a bit more specific here